lib/modules/versioning/hermit/index.ts

```diff
@@ -173,7 +174,14 @@ export class HermitVersioning extends RegExpVersioningApi {
   }
 
   override matches(version: string, range: string): boolean {
-    return this.equals(version, range);
+    if (
+      HermitVersioning._isChannel(version) &&
+      HermitVersioning._isChannel(range)
+    ) {
+      return this.equals(version, range);
+    }
+
+    return satisfies(version, range);
```

Comment:
this will now throw when version is a channel and range isn't and vice versa.

Reply:
ah, you are right about one side channel falls into the`satisfies` call. however I have written some test cases and doesn't see it throw in this method. 

I'll get an other PR up quickly to cover this case.